packages/babel-core/src/transform-ast.ts

```diff
@@ -38,11 +39,11 @@ export const transformFromAst: TransformFromAst = function transformFromAst(
   ast,
   code,
   opts,
-  callback?,
+  callback?: FileResultCallback,
 ) {
   if (typeof opts === "function") {
     callback = opts;
-    opts = undefined;
+    opts = undefined as ValidatedOptions;
```

Comment:
Q: why we need to cast `undefined` as `ValidatedOptions`?

Reply:
Let the type of `opts` be `ValidatedOptions` from the function.

Reply:
Isn't the `opts` parameter typed as `InputOptions | undefined | null,` so `undefined` should work? :thinking:

Reply:
`opts: ValidatedOptions | FileParseCallback`
Both `ValidatedOptions` and `FileParseCallback` can be undefined.